### ISSUE TITLE
Being able to post in multiple languages in the same post.

### ISSUE THREAD

**Issue by Erikcht** [mastodon/mastodon#22766]
### Pitch

Being able to post in multiple languages in the same post.

A checkbox feature or being able to select multiple languages at once before clicking finish.

### Motivation

Today, in order to be able to post in more than one language, on mastodon we have the option to change languages.
The problem is that it only allows me to choose one post at a time.
Mastodon does not allow me to be able to post in English, Spanish and Portuguese at the same time for example.
I know there is an automatic translation feature, but I don't trust these automatic translation software very much, I like to be in control of what I post.
Another thing, many times some influencers are specific to each language, so I could tag some creators in the post in English and some creators in the post in Spanish for example.
[Idea for prototype
](https://i.imgur.com/VBWYCqP.png)

**Comment by velw** [comment]
Seconding this. I often post in English and Japanese, but because the posts can only be in one language I have to write two separate posts, one in each language. It's excessive, when one single post could (in theory, and/or depending on implementation) just be tagged with both languages.

**Comment by Erikcht** [comment]
In the koo microblog, for example, you can have the freedom to post some local sayings since we have that freedom 
![koo](https://user-images.githubusercontent.com/8018995/212792498-ac55c864-2fb2-4f2e-a943-406bac9f5f40.png)
![kooe](https://user-images.githubusercontent.com/8018995/212792502-1b8a9d42-23d5-4363-9eb6-80170390c70a.png)
![kooes](https://user-images.githubusercontent.com/8018995/212792508-825830f7-0249-4620-95fe-1a8666f008d5.png)
[https://www.kooapp.com/koo/Erikcht/df0ebd33-904b-437f-baa8-2b8d5b5fed9f](url)
and even be able to tag different influencers, depending on the language
![koopt1](https://user-images.githubusercontent.com/8018995/212792832-b2ae092f-4610-4deb-8f12-593f864d8f17.png)
![kooen1](https://user-images.githubusercontent.com/8018995/212792835-eafe88c4-10a3-4170-a0f0-c3497414c73a.png)
![kooes1](https://user-images.githubusercontent.com/8018995/212792837-2925d7e0-804b-4e1c-b29e-4a60f7b97bc2.png)
[https://www.kooapp.com/koo/Erikcht/06ba0fc5-2e33-4574-b0af-5cccfbdad5c6](url)

**Comment by ChasBelov** [comment]
I posted a duplicate of this issue, #23698 which I have now closed as duplicate. Copying my proposal as to how the user interface would work:

Steps to reproduce the problem

    Visit ohai.social in English
    Click in the What's on your mind? field.
    Type:

The following should be translated as "Did you have boba?"

有冇波霸奶茶?

The following would reasonably be colloquially translated as "No, and I'm miserable!"

冇囉!

    Select 有冇波霸奶茶?
    In the UI footer, click EN
    In the search box which pops up, type Chinese
    Click 中文 (Chinese)
    Select 冇囉!

Expected behaviour

Current language in the UI displays as EN, and I can proceed to set 冇囉! to ZH as well. When I subsequently post, most of the post will not be wrapped in a tag having the lang="ZH" attribute, only "有冇波霸奶茶?" and "冇囉!" will.
Actual behaviour

Current language in the UI displays as ZH. If I post now, the entire toot will be wrapped in a tag having the lang="ZH" attribute.
Detailed description

WCAG 2.1 has [3.1.2 Language of Parts (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/language-of-parts.html), which states that all content not in the main page's language must be marked with a language attribute.

This is not possible to satisfy for a mixed-language toot. There is a language marker facility, but it can only be applied to the entire toot.

The correct accessible Mastodon behavior would be to support such markup within a toot.

Discussion:

Technically, this is also an issue for mixed-language alternative text attributes, title attributes, and aria-label attributes. However, since HTML does not support language markup within HTML tag attributes, there is nothing Mastodon can do as of Feb. 18, 2023, to remedy this until such time HTML is updated to support Language of Parts within attributes.

Note that the Chinese text is colloquial Cantonese and would best be marked as YUE or ZH-YUE rather than ZH, but that's a feature request and separate from this current issue.
